instrumentation/src/main/java/io/opentelemetry/android/features/diskbuffering/scheduler/DefaultExportScheduler.kt

```diff
@@ -14,11 +18,20 @@ class DefaultExportScheduler : PeriodicRunnable() {
     }
 
     override fun onRun() {
-        // TODO for next PR.
+        val exporter = SignalDiskExporter.get() ?: return
+
+        try {
+            var didExport = exporter.exportBatchOfEach()
+            while (didExport) {
+                didExport = exporter.exportBatchOfEach()
+            }
```

Comment:
I wonder if this can ever be an infinite loop due to a bug, not a big fan of such a while (`possibleAlwaysTrue`) but if there are tests in place, all good.

Reply:
The one reason I see it might run for a long time is if there are a lot of signals in the disk and the internet connectivity is very slow, so each loop might take some time which would cause the whole data to take quite long to finish getting exported, but it shouldn't run forever as `exportBatchOfEach` only returns true if there's data available and it also gets exported successfully, so it's actually quite difficult for this loop to run uninterrupted.